exonum/src/runtime/mod.rs

```diff
@@ -341,18 +357,25 @@ pub trait Runtime: Send + fmt::Debug + 'static {
     /// on the genesis block. Thus, using some core APIs (like requesting the current
     /// blockchain height) will result in a panic.
     ///
+    /// `spec` is the instance specification that used during the `initiate_starting_service`
+    ///  execution.
+    ///
+    /// `status` is the resulting status of the service instance.
+    ///
     /// # Return value
     ///
     /// An error or panic returned from this method will not be processed and will lead
     /// to the node stopping. A runtime should only return an error / panic if the error is local
     /// to the node with reasonable certainty, rather than common to all nodes in the network.
-    /// (The latter kind of errors should be produced during the preceding `start_adding_service`
-    /// call.) The error should contain a description allowing the node administrator to determine
-    /// the root cause of the error and (ideally) recover the node by eliminating it.
-    fn commit_service(
+    /// (The latter kind of errors should be produced during the preceding
+    /// `initiate_adding_service` call.) The error should contain a description allowing the node
+    /// administrator to determine the root cause of the error and (ideally) recover the node by
+    /// eliminating it.
+    fn commit_service_status(
         &mut self,
         snapshot: &dyn Snapshot,
         spec: &InstanceSpec,
+        status: InstanceStatus,
     ) -> Result<(), ExecutionError>;
```

Comment:
Should we continue to return an error here, or ~pull the plug~ just advise to panic? Tbh, I struggle to find a real use case for returning an error.

Reply:
In theory, looking far-far-far away to the future, I can imagine splitting "stopped" service state into two different states: "maintenance", used for e.g. migration to the new version, and "suspended", used e.g. for emulation of "trading hours" for a stock service. In that sense, there will be some "unsuspendable" services like `Supervisor`.

But currently I don't see a need in returning an error here as well.